test/annotation_utils.t

```diff
@@ -864,6 +873,170 @@ subtest '_run_rast_workflow_ann' => sub {
+        else {
+            print "\nThere is NO contigID_hash in $pgn->{id}:\n".Dumper(keys %rd);
+        }
+        ## Run the remapping
+        my $pgn1 = $annoutil->_remap_contigIDs( $ctgID_hash, $pgn );
+        print "\n=========after remapping the rasted genome on $pgn->{id}:\n";
+
+        if( $pgn1->{contig_ids} &&  @{$pgn1->{contig_ids}} > 0) {
+            print "\n------the rasted genome's contig_ids:\n".
+                   Dumper( $pgn1->{contig_ids});
+            my $exp_ctg_ids = $pgn->{contig_ids};
+            for my $cid (@{$exp_ctg_ids}) {
+                $cid = $ctgID_hash->{$cid};
+            }
+            cmp_deeply $pgn1->{contig_ids}, $exp_ctg_ids,
+                        'contig ids remapped correctly';
+        }
+        if( $pgn1->{contigs} &&  @{$pgn1->{contigs}} > 0) {
+            for( my $i = 0; $i < @{$pgn1->{contigs}}; $i++ ) {
+                my $ctg_id = @{$pgn1->{contigs}}[$i]->{id};
+                my $old_ctg_id = @{$pgn->{contigs}}[$i]->{id};
+                if( $i < $cnt_limit ) {
+                  print "\n------the rasted genome contig's id=$ctg_id";
+                }
+                ok( (!defined($ctgID_hash->{$old_ctg_id}) && $ctg_id eq $old_ctg_id) ||
+                     $ctg_id eq $ctgID_hash->{$old_ctg_id},
+                    'id of contigs re-mapped correctly' );
+            }
+        }
+        if( $pgn1->{features} && @{$pgn1->{features}} > 0) {
+            for( my $i = 0; $i < @{$pgn1->{features}}; $i++ ) {
+                my $ftr_locid = @{$pgn1->{features}}[$i]->{location}[0][0];
+                my $old_ftr_locid = @{$pgn->{features}}[$i]->{location}[0][0];
+                if( $i < $cnt_limit ) {
+                  print "\n------the rasted genome feature location's contig_id=".
+                        "$ftr_locid";
+                }
+                ok( (!defined($ctgID_hash->{$old_ftr_locid}) &&
+                      $ftr_locid eq $old_ftr_locid) ||
+                      $ftr_locid eq $ctgID_hash->{$old_ftr_locid},
+                    'id of contigs in feature locations re-mapped correctly' );
+            }
+        }
+        if( $pgn1->{non_coding_features} &&
+            @{$pgn1->{non_coding_features}} > 0) {
+            for( my $i = 0; $i < @{$pgn1->{non_coding_features}}; $i++ ) {
+                my $nc_ftr_locid = @{$pgn1->{non_coding_features}}[$i]->{location}[0][0];
+                my $old_ncftr_locid = @{$pgn->{non_coding_features}}[$i]->{location}[0][0];
+                if( $i < $cnt_limit ) {
+                  print "\n------the rasted genome nc_feature location's contig_id=".
+                        "$nc_ftr_locid";
+                }
+                ok( (!defined($ctgID_hash->{$old_ncftr_locid}) &&
+                      $nc_ftr_locid eq $old_ncftr_locid) ||
+                      $nc_ftr_locid eq $ctgID_hash->{$old_ncftr_locid},
+                    'id of contigs in nc_feature locations re-mapped correctly' );
+            }
+        }
```

Comment:
This and the following code blocks could be refactored as they repeat the structure of the previous block.

Reply:
rewritten